### ISSUE TITLE
Text panel: Link to similar path on same domain going to wrong place

### ISSUE THREAD

**Issue by mkopinsky** [grafana/grafana#36619]
**What happened**:

I have a text panel on my home dashboard with these contents:
```markdown
# Grafana environments

* [Prod](https://labs.waytohealth.org/grafana/)
* [Staging](https://labs.waytohealth.org/grafana-stg/)
* [Test](https://labs.waytohealth.org/grafana-tst/)
```

From prod, if I click on the staging link, it takes me to https://labs.waytohealth.org/grafana/-stg/ (which 404s) instead of https://labs.waytohealth.org/grafana-stg/. Right click and open in new tab works fine. I'm guessing this is angular router trying to be too smart for its own good?

**How to reproduce it (as minimally and precisely as possible)**:

* Create a grafana instance with a root URL of http://example.com/foo
* Create a second grafana instance with a root URL of http://example.com/foo-bar (I guess this doesn't have to be a grafana instance, it could be anything else hosted at that path
* Try to link from the first instance to the second via a Text panel

**Anything else we need to know?**:

Relevant config setting:
```yaml
    environment:
      - GF_SERVER_DOMAIN=labs.waytohealth.org
      - GF_SERVER_ROOT_URL=https://labs.waytohealth.org/grafana/
      - GF_SERVER_SERVE_FROM_SUB_PATH=true
```
and that does show up in Settings with the trailing slash

**Environment**:
- Grafana version: v8.0.2 (5c5d45e19f)
- Data source type & version: N/A
- OS Grafana is installed on: default docker image
- User OS & Browser: Chrome 91.0.4472.114
- Grafana plugins: no custom plugins, whatever is in base docker image
- Others:

**Comment by torkelo** [comment]
Fixed in v8.0.5 https://github.com/grafana/grafana/pull/36498

**Comment by mkopinsky** [comment]
This is still happening in 8.0.5 (cbb2aa5001)
